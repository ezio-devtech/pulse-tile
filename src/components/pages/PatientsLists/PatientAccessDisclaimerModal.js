import React from 'react';
import PropTypes from 'prop-types';
import { Modal } from 'react-bootstrap';

const PatientAccessDisclaimerModal = ({ onAgreeRedirectTo, onClose, ...restProps }) => {
  const handleAgree = () => restProps.history.push(onAgreeRedirectTo);

  return (
    <Modal.Dialog>
      <div className="panel panel-secondary without-margin">
        <div className="panel-heading">
          <h3 className="panel-title">Patient Access Disclaimer</h3>
        </div>

        <div className="panel-body">
          <div className="panel-body-inner">
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed nec lobortis elit. Aenean mi nunc, feugiat ut aliquet non, iaculis vel tellus. Donec semper felis placerat, posuere nisi a, suscipit turpis. Integer sit amet lacus pellentesque, vestibulum libero id, sagittis nisi. Phasellus eleifend, neque eget vulputate semper, enim dui dictum neque, non iaculis felis augue at nunc.
          </div>
          <div className="panel-control">
            <div className="wrap-control-group hide-indent-bottom">
              <div className="control-group with-indent right">
                <button className="btn btn-danger btn-icon-normal" onClick={onClose}><i className="btn-icon fa fa-ban" /> <span className="btn-text">Decline</span></button>
                <button className="btn btn-success" onClick={handleAgree}>
                  <span className="btn-text">Agree</span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </Modal.Dialog>
  )
};

PatientAccessDisclaimerModal.propTypes = {
  onClose: PropTypes.func.isRequired,
  onAgreeRedirectTo: PropTypes.string.isRequired,
  history: PropTypes.shape({
    push: PropTypes.funct,
  }).isRequired,
};

export default PatientAccessDisclaimerModal;
