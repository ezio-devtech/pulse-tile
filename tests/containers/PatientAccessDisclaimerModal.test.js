import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';

import PatientAccessDisclaimerModal from '../../src/components/containers/PatientsList/PatientAccessDisclaimerModal';

Enzyme.configure({ adapter: new Adapter() });

const history = {
  push: () => {},
};

describe('Component <PatientAccessDisclaimerModal />', () => {
  it('should renders mount correctly', () => {
    const patientAccessDisclaimerModal = mount(
      <PatientAccessDisclaimerModal
        history={history}
      />
    );
    expect(patientAccessDisclaimerModal).toMatchSnapshot();
  });
  it('should renders shallow correctly', () => {
    const patientAccessDisclaimerModal = shallow(
      <PatientAccessDisclaimerModal
        history={history}
      />
    );
    patientAccessDisclaimerModal.find('.btn-success').simulate('click');
    expect(patientAccessDisclaimerModal).toMatchSnapshot();
  });
});

