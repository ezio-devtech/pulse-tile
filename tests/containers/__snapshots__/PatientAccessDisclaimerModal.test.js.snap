// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <PatientAccessDisclaimerModal /> should renders mount correctly 1`] = `
<PatientAccessDisclaimerModal
  history={
    Object {
      "push": [Function],
    }
  }
>
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    dialogComponentClass={[Function]}
    enforceFocus={true}
    keyboard={true}
    manager={
      ModalManager {
        "add": [Function],
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "isTopModal": [Function],
        "modals": Array [],
        "remove": [Function],
      }
    }
    onHide={[Function]}
    renderBackdrop={[Function]}
    restoreFocus={true}
    show={false}
  >
    <Modal
      autoFocus={true}
      backdrop={true}
      backdropClassName="modal-backdrop"
      backdropTransitionTimeout={150}
      containerClassName="modal-open"
      dialogTransitionTimeout={300}
      enforceFocus={true}
      keyboard={true}
      manager={
        ModalManager {
          "add": [Function],
          "containers": Array [],
          "data": Array [],
          "handleContainerOverflow": true,
          "hideSiblingNodes": true,
          "isTopModal": [Function],
          "modals": Array [],
          "remove": [Function],
        }
      }
      onEntering={[Function]}
      onExited={[Function]}
      onHide={[Function]}
      renderBackdrop={[Function]}
      restoreFocus={true}
      show={false}
      transition={[Function]}
    />
  </Modal>
</PatientAccessDisclaimerModal>
`;

exports[`Component <PatientAccessDisclaimerModal /> should renders shallow correctly 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={false}
>
  <div
    className="panel panel-secondary without-margin"
  >
    <div
      className="panel-heading"
    >
      <h3
        className="panel-title"
      >
        Patient Access Disclaimer
      </h3>
    </div>
    <div
      className="panel-body"
    >
      <div
        className="panel-body-inner"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed nec lobortis elit. Aenean mi nunc, feugiat ut aliquet non, iaculis vel tellus. Donec semper felis placerat, posuere nisi a, suscipit turpis. Integer sit amet lacus pellentesque, vestibulum libero id, sagittis nisi. Phasellus eleifend, neque eget vulputate semper, enim dui dictum neque, non iaculis felis augue at nunc.
      </div>
      <div
        className="panel-control"
      >
        <div
          className="wrap-control-group hide-indent-bottom"
        >
          <div
            className="control-group with-indent right"
          >
            <button
              className="btn btn-danger btn-icon-normal"
            >
              <i
                className="btn-icon fa fa-ban"
              />
               
              <span
                className="btn-text"
              >
                Decline
              </span>
            </button>
            <button
              className="btn btn-success"
              onClick={[Function]}
            >
              <span
                className="btn-text"
              >
                Agree
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</Modal>
`;
